### DISCUSSION TITLE
Is there a lint configuration that would deny usage of std::any::Any?

### DISCUSSION THREAD

**Discussion by facetious** [rust-lang/rust-clippy#13451] [Q&A] [answered]
I'm interested in denying the local usage of `Any` in my project's crates. I don't mind `Any` usage in third-party crates that are dependencies of the project, but if at all possible, I would love for a lint to be able to error or warn on `Any` usage. Compare to eslint's `no-explicit-any` rule, where a developer must explicitly disable the rule (and provide justification to their peers) in order to use `Any`. Rust's type system is very nice, and with a project that is intended to be closed and finite, there's little justification for the usage of `Any`.

Is there any combination of lints or configurations that I could use to trigger this behaviour?

**Comment by flip1995** [comment, marked as answer]
There's no such lint currently. If you want, we can add such a lint as an allow-by-default (opt-in) lint in the `restriction` group. If you want to contribute this lint, take a look at our book and how to add new lints: https://doc.rust-lang.org/clippy/development/defining_lints.html

> with a project that is intended to be closed and finite

In Rust there is almost never a reason why you should use `Any`, even during development. I've never needed it in the >6 years I've been writing Rust. I've also never seen `Any` in other code, when it wasn't absolutely necessary. Probably the reason, why this lint was never requested or implemented.

**Reply by facetious** [reply]
> I've never needed it in the >6 years I've been writing Rust. I've also never seen `Any` in other code, when it wasn't absolutely necessary.

Interestingly, that's the main motivator for _why_ I'm requesting it. :)

It's easier to have a lint double-check work before it hits peer review. I'll add your link to my TODO list and see about contributing back.
